### DISCUSSION TITLE
tutorials/xr/setting_up_xr

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#23] [User-contributed notes]
# tutorials/xr/setting_up_xr

Introduction to the XR system in Godot: Godot provides a modular XR system that abstracts many of the different XR platform specifics away from the user. At the core sits the XRServer which acts as...

https://docs.godotengine.org/en/latest/tutorials/xr/setting_up_xr.html

**Comment by RaphaelRoyerRivard** [comment]
It's worth noting that if you are using monado as the OpenXR implementation, you will need to run the command `monado-service` in a terminal for Godot to detect your VR headset.

**Comment by LillyPK** [comment]
when using gd-script, everything works as intended but the C# implementation will show a grey screen on the headset and displays what you should be seeing on the headset in a window

**Reply by Calinou** [reply]
To my knowledge, there's nothing preventing XR from working with C#. I suggest asking on other community platforms (and making sure you're using Godot 4.3 if you haven't upgraded already): https://godotengine.org/community

**Reply by GIJaws** [reply]
I'm having the same issue as @LillyPK

**Reply by GIJaws** [reply]
@LillyPK got it working by changing the class name of C# script to Main and I also set up Visual Studio code as an external editor for Godot.

**Comment by saimukzi** [comment]
In order to see from XRCamera3D, set Reference Space = Local Floor, refer to https://docs.godotengine.org/en/stable/tutorials/xr/openxr_settings.html .

**Comment by FPChris** [comment]
Why is the Compatibility renderer suggested for Quest 3? If you use the Godot Android Editor inside a Quest 3 headset it defaults to Mobile and runs everything just fine.

**Reply by Calinou** [reply]
When the page was initially written, Mobile wasn't as optimized as it is now. That said, Compatibility is still expected to run better in most scenarios, especially in simpler projects where it has a lower base cost. The Compatibility renderer doesn't have as many features as Mobile though.
